src/webpack-benchmark.js

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+const webpack = require("webpack");
+
+const payloads = [
+  {
+    entry: "/src/fixtures/webpack/a.js"
+  }
+].map(({ entry }, i) => ({
+  entry,
+  output: {
+    path: "/dist/webpack/",
+    filename: `bundle.${i}.js`
+  },
+  // Using bail option in order to receive noisy errors in the benchmark if something went wrong
+  bail: true,
+  // We need to define that because Firefox has a Object.prototype.watch function
+  watch: false
+}));
+
+module.exports = {
+  name: "webpack",
+  defer: true,
+  fn(deferred) {
+    return Promise.all(
```

Comment:
This is a problem. I carefully avoided any interaction with the microtask queue, i.e. any use of promises, because that way you no longer measure the core JavaScript bits, but all of a sudden mix in embedder tasks into the measurement (i.e. DOM tasks scheduled on the microtask queue). Also the microtask queue doesn't work the same across different shells (not even across different versions of the same shell).

Please use a synchronous API instead.

Ideally webpack (and rollup FWIW) would provide a simple API entry where you can feed in strings and it would do the tree shaking and bundling on that, and produce a single string output.

Reply:
Ok, that is going to be interesting 😁 maybe it's possible, but I'm not sure how deep the assumption of an async file system access is buried in the code 😁 

We will also leave the regular code path, but that's probably a fair trade-off.

Reply:
I think that's the only option for this, otherwise we get all kinds of other stuff into the measurement and the results will vary based on whether it's run in Node or the browser.

Reply:
Ok, I took a look and so far it looks promising.

Basically, we just need to provide synchronous mocks for `setImmediate()` and `process.nextTick()`. There are, however, assumptions about the execution order. That's why I had to implement a simple event loop. But with my event loop, it looks like webpack is compiling synchronously. The event loop looks basically like this:

```js
const loop = [];

global.process = {
  nextTick(fn) {
    const args = Array.prototype.slice.call(arguments, 1);
    loop.push(() => {
      fn.apply(null, args);
    });
  }
};
global.setImmediate = function (fn) {
  const args = Array.prototype.slice.call(arguments, 1);
  loop.push(() => {
    fn.apply(null, args);
  });
};

// and then during the benchmark

let task;
while ((task = loop.shift())) {
  task();
}
```

Do you think that would be ok?

Reply:
Interesting approach. Do you think we could somehow make the `setImmediate` and `process.nextTick` mock local to the webpack benchmark? I don't like changing global state. That can easily screw up other benchmarks w/o us noticing it.

Reply:
Yes, we could inject our `process` and `setImmediate` mock to only webpack and virtualfs  (virtualfs is using `setImmediate` to fake async method calls).

However, I would inject these mocks to *all* modules. Because if any module inside the benchmark is using one of these methods, the benchmark cannot be sync by design so I'd say it's fair to mock these throughout the whole bundle.

I would inject these mocks into the local module scope instead of mutating `global` which ensures that we don't touch the global object.